src/pages/billing/_tests/CreateAzureBillingProjectModal.test.js

```diff
@@ -0,0 +1,239 @@
+              ]
+            }
+          ),
+          createAzureProject: (billingProjectName, tenantId, subscriptionId, managedResourceGroupId) => {
+            createResult.billingProjectName = billingProjectName
+            createResult.tenantId = tenantId
+            createResult.subscriptionId = subscriptionId
+            createResult.managedResourceGroupId = managedResourceGroupId
+            Promise.resolve(createResult)
+          }
+        }
+      }
+    })
+
+    // Act - Insert valid billing project name.
+    await act(async () => {
+      await userEvent.type(getBillingProjectInput(), projectName)
+    })
+    // Assert
+    verifyDisabled(getCreateButton())
+
+    // Act - Supply valid subscription UUID and wait for Ajax response
+    await act(async () => {
+      await userEvent.type(getSubscriptionInput(), uuid())
+    })
+    await waitFor(() => verifyEnabled(getManagedAppInput()))
+    // Assert
+    verifyDisabled(getCreateButton())
+
+    // Act - Select one of the managed apps
+    await userEvent.click(getManagedAppInput())
+    const selectOption = await screen.findByText(appName)
+    await userEvent.click(selectOption)
+    // Assert
+    verifyEnabled(getCreateButton())
+
+    // Act - Click Create
+    await act(async () => {
+      await userEvent.click(getCreateButton())
+    })
+    // Assert - Verify Ajax billing project creation function was called.
+    expect(createResult.billingProjectName).toBe(projectName)
+    expect(createResult.tenantId).toBe(tenant)
+    expect(createResult.subscriptionId).toBe(subscription)
+    expect(createResult.managedResourceGroupId).toBe(mrg)
+  })
+
+  it('shows an error if billing project name is not unique', async () => {
+    // Arrange - Mock managed app Ajax call to succeed, but create billing project call to return duplicate error.
+    const projectName = 'Dupe_Billing_Project_Name'
+    const appName = 'appName'
+    Ajax.mockImplementation(() => {
+      return {
+        Billing: {
+          listAzureManagedApplications: () => Promise.resolve(
+            {
+              managedApps: [
+                { applicationDeploymentName: appName, tenantId: 'fakeTenant', subscriptionId: 'fakeSub', managedResourceGroupId: 'fakeMrg' }
+              ]
+            }
+          ),
+          createAzureProject: () => Promise.reject(new Response('Mock duplicate error', { status: 409 }))
+        }
+      }
+    })
+
+    // Act - Insert valid billing project name and subscription UUID, select the managed app.
+    await act(async () => {
+      await userEvent.type(getBillingProjectInput(), projectName)
+      await userEvent.type(getSubscriptionInput(), uuid())
+    })
+    await waitFor(() => verifyEnabled(getManagedAppInput()))
+    // Select one of the managed apps
+    await userEvent.click(getManagedAppInput())
+    const selectOption = await screen.findByText(appName)
+    await userEvent.click(selectOption)
+
+    // Assert
```

Comment:
I fail to see how comments like this one add enough value to pay for the vertical space used.